sqlglot/parser.py

```diff
@@ -5911,3 +5919,50 @@ def _replace_lambda(
                     else:
                         column.replace(dot_or_id)
         return node
+
+    def _parse_truncate(self) -> t.Optional[exp.Truncate] | exp.Expression:
+        # Not to be confused with TRUNCATE(number, decimals) function call
+        if self._match(TokenType.L_PAREN):
+            return self._parse_function_call(anonymous=True)
+
+        # Clickhouse supports TRUNCATE DATABASE as well
+        is_database = self._match(TokenType.DATABASE)
+
+        self._match(TokenType.TABLE)
+
+        exists = self._parse_exists(not_=False)
+
+        def _parse_truncate_tables() -> t.Optional[exp.Expression]:
+            this = self._parse_table(schema=True, is_db_reference=is_database)
```

Comment:
Curious about the use of `schema=True` here - we shouldn't be expecting a schema, right?
```suggestion
            this = self._parse_table(is_db_reference=is_database)
```

Reply:
Ah, we _do_ need `schema=True` here otherwise we'll try to parse a table alias etc. Your implementation is correct, apologies for the misleading comment.